alacritty/src/event.rs

```diff
@@ -479,6 +479,7 @@ impl<N: Notify> Processor<N> {
                     | Destroyed
                     | HoveredFile(_)
                     | Touch(_)
+                    | Focused(_)
```

Comment:
Oh, my bad. I didn't think about this. Of course with a guard we'd have to handle this separately and we don't have a catch-all.

Since this handles just a single event, we could also go for the early return, that would not require additional indentation and save us the `Focused(_)` catch-all kinda handling.

Something like this:

```rust
Focused(is_focused) => {
  if window_id != processor.ctx.window.window_id() {
    return;
  }
  // ...
}
```

I think this might be our best option for code style. Though curious what you think.

Reply:
This was my initial code, then I've moved to "if true { }" to avoid `return` and because we'd already had something similar in `MouseInput`/`KeyboardInput`, option with match guard wasn't considered at all because of the exact same thing you've reject it now btw.

Reply:
I'm fine with the option you've suggested here " if != then return "

Reply:
Honestly, I think you're right and avoiding a `return` makes sense, because having returns everywhere is much more difficult to read than just branching (since we don't know if it skips some other part of the method). So the cognitive complexity would be lower without the return and I don't think the indentation is a super big deal.

I didn't think of the issue with the match guard, but I think that makes things pretty ugly, so I'd rather stay away from that.

So sorry for the noise, I think your initial approach was probably the best solution already and I think your points do make sense. Want to go with that then?

Reply:
Sure